.pre-commit-config.yaml

```
@@ -42,12 +45,11 @@ repos:
   #   rev: v1.6.0
   #   hooks:
   #     - id: mypy
-  # - repo: local
-  #   hooks:
-  #     - id: pylint
-  #       name: pylint
-  #       entry: bash -c 'test -d .venv && . .venv/bin/activate ; pylint ${CI:+--reports=yes} "$@"' -
-  #       language: system
-  #       types: [ python ]
-  #       args:
-  #         - --disable=R,C
+  - repo: local
+    hooks:
+      - id: pylint
+        name: pylint
+        entry: bash -c 'test -d .venv && . .venv/bin/activate ; pylint ${CI:+--reports=yes} "$@"' -
```

```suggestion
        entry: pylint
```
Not sure why this is necessary, pylint ignore venv by default I think (Or we can ignore it in pylintrc)

---

This is coming from another CI 😅 

It is saying "using bash: if there is a `.venv`, load it `;` run `pylint` with `$@`" (it's a bit more nuanced, but that's the gist).

`.venv` is my go-to name for venv locally, and pylint naturally needs to be installed in the venv to run.

---

It"s not super important, just felt like we already modify the --report option elsewhere. Could be nice to enter the venv automatically, I generally let the user place themself in the venv before launching pre-commit but it's an hassle of course.

---

I know that `.venv` is me-specific, but I haven't figured out a way yet to make it more generic. 😅 

However, that does not preclude user entering her own venv - just gives a hand to `.venv`-ers

---

Oh, and the

> just felt like we already modify the --report option elsewhere

The "consensus" behind this copy-paste is: People don't care about `--report`, but we can show it in CI.

---

But pre-commit launch multiple instance of pre-commit on a subset of the file so the report will get printed multiple time (also the CI is pre-commit itself). So I added the option to not have the reports/score in ``args``
```suggestion
        entry: bash -c 'test -d .venv && . .venv/bin/activate ; pylint "$@"' -
```

---

Should I apply the suggestion above ?

---

Yeeeeah, I guess `-sn` `-rn` can come up and drop `${CI:+--reports=yes}`.

But spend minimal time on it 😅

---

Done, I think it's mergeable now :)